packages/expo-notifications/src/scheduleNotificationAsync.ts

```diff
@@ -29,6 +29,12 @@ function parseTrigger(userFacingTrigger: NotificationTriggerInput): NativeNotifi
     return { type: 'date', timestamp: userFacingTrigger.getTime() };
   } else if (typeof userFacingTrigger === 'number') {
     return { type: 'date', timestamp: userFacingTrigger };
+  } else if ('hour' in userFacingTrigger && 'minute' in userFacingTrigger) {
+    return {
+      type: 'dailyInterval',
+      hour: userFacingTrigger.hour ?? 0,
```

Comment:
Yes, I think it would be good to move the input validation logic here and raise an error if it's not within the bounds. That way it will be consistent and doesn't have to be duplicated on the native sides.